src/abstractdataframe/selection.jl

```diff
@@ -179,9 +199,12 @@ and transformed using the `old_column => fun => new_column_name` syntax.
 is a `Symbol` or an integer then `fun` is applied to the corresponding column vector.
 Otherwise `old_column` can be any column indexing syntax, in which case `fun`
 will be passed the column vectors specified by `old_column` as separate arguments.
-If `fun` returns a value of type other than `AbstractVector` then it will be wrapped
-into a 1-element vector, unless its type is one of `AbstractDataFrame`, `NamedTuple`,
-`DataFrameRow`, `AbstractMatrix`, in which case an error is thrown as currently these
+If `fun` returns a value of type other than `AbstractVector` then: it will be spread
+into a vector matching the target number of rows in the data frame
+(as a particular rule a values stored in a `Ref` or a `0`-dimensional `AbstractArray`
+ are unwrapped and treated in the same way),
+unless its type is one of `AbstractDataFrame`, `NamedTuple`, `DataFrameRow`,
+`AbstractMatrix`, in which case an error is thrown as currently these
 return types are not allowed.
```

Comment:
```suggestion
If `fun` returns a value of type other than `AbstractVector` then it will be broadcasted
into a vector matching the target number of rows in the data frame,
unless its type is one of `AbstractDataFrame`, `NamedTuple`, `DataFrameRow`,
`AbstractMatrix`, in which case an error is thrown as currently these
return types are not allowed.
As a particular rule, values wrapped in a `Ref` or a `0`-dimensional `AbstractArray`
are unwrapped and then broadcasted.
```